tensorboard/webapp/metrics/views/card_renderer/scalar_card_container.ts

```diff
@@ -342,7 +340,31 @@ export class ScalarCardContainer implements CardRenderer, OnInit {
 
           return {seriesId: partial.seriesId, points: normalizedPoints};
         });
-      }),
+      })
+    );
+
+    const partitionedSeries$ = normalizedSeries$.pipe(
+      combineLatestWith(
+        this.store.select(getMetricsScalarPartitionNonMonotonicX)
+      ),
+      map<[PartialSeries[], boolean], PartitionedSeries[]>(
+        ([normalizedSeries, enablePartition]) => {
+          if (enablePartition) return partitionSeries(normalizedSeries);
+
+          return normalizedSeries.map((series) => {
+            return {
+              ...series,
+              runId: series.seriesId,
```

Comment:
Upon first read, I think it is a little non-obvious that converting `seriesId` back into `runId` is valid.  It is valid here, because `normalizedSeries$` returns an observable of `PartialSeries`, and we are implicitly guaranteeing that seriesId === runId for `PartialSeries` (but not for `PartitionedSeries`).

See comment below.